paz/models/detection/frustum_pointnet.py

```diff
@@ -0,0 +1,177 @@
+import tensorflow as tf
+from tensorflow.keras.layers import Conv2D, Dense
+from tensorflow.keras.layers import Dropout
+from tensorflow.keras.layers import Input, Lambda
+from tensorflow.keras.layers import MaxPooling2D
+from tensorflow.keras.models import Model
+
+from paz.models.detection.model_util import NUM_HEADING_BIN, NUM_SIZE_CLUSTER
+from paz.models.detection.model_util import parse_output_to_tensors
+from paz.models.detection.model_util import point_cloud_masking
+from paz.optimization.losses.frustumpointnet_loss import FPointNet_loss
+
+
+def Instance_Seg_Net(point_cloud, one_hot_vec):
+
+    num_points = point_cloud.get_shape().as_list()[1]
+
+    net = tf.expand_dims(point_cloud, 2)
+
+    net = Conv2D(64, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_1')(net)
+
+    net = Conv2D(64, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_2')(net)
+
+    point_feat = Conv2D(64, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_3')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_4')(
+        point_feat)
+
+    net = Conv2D(1024, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_5')(net)
+
+    global_feat = MaxPooling2D(pool_size=[num_points, 1], padding='VALID')(net)
+
+    global_feat = tf.concat([global_feat, tf.expand_dims(tf.expand_dims(one_hot_vec, 1), 1)], axis=3)
+    global_feat_expand = tf.tile(global_feat, [1, num_points, 1, 1])
+    concat_feat = tf.concat(axis=3, values=[point_feat, global_feat_expand])
+
+    net = Conv2D(512, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_6')(
+        concat_feat)
+
+    net = Conv2D(256, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_7')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_8')(net)
+
+    net = Conv2D(128, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation='relu', name='conv1_9')(net)
+
+    net = Dropout(rate=0.5)(net)
+
+    logits = Conv2D(2, kernel_size=[1, 1], padding='VALID', strides=[1, 1], activation=None, name='conv1_10')(net)
+
+    logits = tf.squeeze(logits, [2])  # BxNxC
+
+    return logits
+
+
+def Box_Est_Net(object_point_cloud, one_hot_vec):
```

Comment:
use full camel case name for naming a model

Reply:
Done